src/main/java/com/suchocki/parkingmeter/dao/ParkActionDAOImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.suchocki.parkingmeter.dao;
+
+import java.util.List;
+
+import org.springframework.stereotype.Repository;
+import com.suchocki.parkingmeter.database.FakeDatabaseStub;
+import com.suchocki.parkingmeter.entity.Driver;
+import com.suchocki.parkingmeter.entity.ParkAction;
+
+@Repository
+public class ParkActionDAOImpl implements ParkActionDAO {
+
+	@Override
+	public void save(ParkAction parkAction) {
+		for (ParkAction pAction : FakeDatabaseStub.parkActions) {
+			if (pAction.getId() == parkAction.getId()) {
+				update(parkAction);
+			}
+		}
+		parkAction.setId(FakeDatabaseStub.parkActions.size() + 1);
+		FakeDatabaseStub.parkActions.add(parkAction);
+	}
+
+	@Override
+	public ParkAction get(Integer id) {
```

Comment:
daczego duży `Integer`?

Reply:
Ze względu na generyczność interfejsu _DAO_. Nie można by było podać _int_ jako typ identyfikatora. _ParkActionDAOImpl_ implementuje interfejs _ParkActionDAO_, który rozszerza _DAO_.
Ewentualnie, _ParkActionDAO_ mógłby nie rozszerzać _DAO_, ale uznałem, że tak będzie wygodniej. 
Choć oczywiście tworzy to zagrożenie przekazania _null_ jako argumentu wywołania metody.